include/mesh/mesh_base.h

```
@@ -1820,8 +1830,27 @@ class MeshBase : public ParallelObject
   const std::set<subdomain_id_type> & get_mesh_subdomains() const
   { libmesh_assert(this->is_prepared()); return _mesh_subdomains; }
 
+#ifdef LIBMESH_ENABLE_PERIODIC
+  /**
+   * Register a pair of boundaries as disconnected boundaries.
+   */
+  void add_disconnected_boundaries(const boundary_id_type b1,
+                                   const boundary_id_type b2,
+                                   const RealVectorValue & translation);
+
+  PeriodicBoundaries * get_disconnected_boundaries();
+
+  const PeriodicBoundaries * get_disconnected_boundaries() const;
+#endif
+
+
 protected:
 
+ #ifdef LIBMESH_ENABLE_PERIODIC
+  /// @brief The disconnected boundary id pairs.
+  std::unique_ptr<PeriodicBoundaries> _disconnected_boundary_pairs;
```

And now I finally see something I should have noticed right away, the fact that I *don't* see some things:

We're not assigning these in `operator=` or the copy constructor, and we're not testing for size equality in `operator==` via `locally_equals`.  We definitely need both those things to be done, as a matter of principle, and because MOOSE will happily `clone()` meshes left and right and we don't want to lose `_disconnected_boundary_pairs` or violate sanity checks when it does.

I'd say we should also be doing *something* in `UnstructuredMesh::stitching_helper`, given how heavily MOOSE users rely on that these days.  Probably a `if (have_disc_bdys) libmesh_deprecated_msg()` there is sufficient for now, and users can work around it by adding disconnected boundary pairs only at the end of their mesh generation process, but we can't just silently drop disconnected boundaries from one of two stitched mesh inputs.

---

Thanks, Roy. This is a good catch!
I have implemented:
1. `operator= ` to include `_disconnected_boundary_pairs`.
2. `locally_equals` includes checking for `_disconnected_boundary_pairs`.
3. `stitching_helper ` includes `if (have_disc_bdys) libmesh_deprecated()`.

Thank you for your feedback!

---

Sure, @roystgnr! 😃 
I have changed `libmesh_deprecated` to `libmesh_not_implemented_msg `.
Thanks!